providers/gce/gce.go

```diff
@@ -741,6 +741,11 @@ func (g *Cloud) OnXPN() bool {
 	return g.onXPN
 }
 
+// NetworkName returns the network name
+func (g *Cloud) NetworkName() string {
```

Comment:
Not clear why this is a method rather than use the existing func to deal with resource URLs

k8s-cloud-provider/pkg/cloud ResourceID parses these URLs in general.

Reply:
Are you suggesting creating a function that parses network url with the `ParseResourceURL` function on the consumer side? Or swapping the implementation of the `NetworkName` function with `ParseResourceURL`?

Reply:
My comment is more on whether or not we need to so many methods to get very similar information. For example, having the full URL also has the project identity, which may be used later etc.